specification/imagebuilder/resource-manager/Microsoft.VirtualMachineImages/stable/2024-02-01/imagebuilder.json

```diff
@@ -0,0 +1,2095 @@
+            }
+          }
+        },
+        "provisioningState": {
+          "$ref": "#/definitions/ProvisioningState",
+          "description": "Provisioning state of the resource",
+          "readOnly": true
+        },
+        "provisioningError": {
+          "$ref": "#/definitions/ProvisioningError",
+          "description": "Provisioning error, if any",
+          "readOnly": true
+        },
+        "lastRunStatus": {
+          "$ref": "#/definitions/ImageTemplateLastRunStatus",
+          "description": "State of 'run' that is currently executing or was last executed.",
+          "readOnly": true
+        },
+        "buildTimeoutInMinutes": {
+          "type": "integer",
+          "format": "int32",
+          "minimum": 0,
+          "maximum": 960,
+          "default": 0,
+          "description": "Maximum duration to wait while building the image template (includes all customizations, optimization, validations, and distributions). Omit or specify 0 to use the default (4 hours)."
+        },
+        "vmProfile": {
+          "$ref": "#/definitions/ImageTemplateVmProfile",
+          "description": "Describes how virtual machine is set up to build images"
+        },
+        "stagingResourceGroup": {
+          "type": "string",
+          "description": "The staging resource group id in the same subscription as the image template that will be used to build the image. If this field is empty, a resource group with a random name will be created. If the resource group specified in this field doesn't exist, it will be created with the same name. If the resource group specified exists, it must be empty and in the same region as the image template. The resource group created will be deleted during template deletion if this field is empty or the resource group specified doesn't exist, but if the resource group specified exists the resources created in the resource group will be deleted during template deletion and the resource group itself will remain."
+        },
+        "exactStagingResourceGroup": {
+          "type": "string",
+          "description": "The staging resource group id in the same subscription as the image template that will be used to build the image. This read-only field differs from 'stagingResourceGroup' only if the value specified in the 'stagingResourceGroup' field is empty.",
+          "readOnly": true
+        },
+        "autoRun": {
+          "$ref": "#/definitions/ImageTemplateAutoRun",
+          "description": "Indicates whether or not to automatically run the image template build on template creation or update."
+        },
+        "managedResourceTags": {
+          "type": "object",
+          "additionalProperties": {
```

Comment:
>additionalProperties [](http://example.com/codeflow?start=11&length=20)

This is an ARM anti-pattern and generally not allowed. Can you schematize these or switch to an array?

Reply:
Hi @mentat9 this is a user-defined tag that is supposed to be allowed: 
![image](https://github.com/Azure/azure-rest-api-specs/assets/90284189/327e4b06-5070-4d93-8678-6f3e2b30daac)


cc: @feiyushi

Reply:
Why is this needed in addition to the top level "tags" property that is already defined for tracked resources?

Reply:
the service will create additional vm and related resources in a resource group. this tag is to allow customer to tag those resources. the top-level tags only applied to image template resource itself.

Reply:
@mentat9 can we get this PR signed off?

Reply:
Can you explain this in a little more detail? It's not clear to me why your API is passing tags through but no other properties (like resources, RG name, location, etc). It seems to be blurring the line between resources managed by your RP and those managed by the customer. If these resources are visible/accessible to the customer, they can tag the RG themselves. If they are not visible/accessible, then what good is tagging them?

Reply:
If you believe your scenario requires the use of additionalProperties please book an ARM modeling office hours slot to explain the reasoning

Reply:
> Can you explain this in a little more detail? It's not clear to me why your API is passing tags through but no other properties (like resources, RG name, location, etc). It seems to be blurring the line between resources managed by your RP and those managed by the customer. If these resources are visible/accessible to the customer, they can tag the RG themselves. If they are not visible/accessible, then what good is tagging them?

FYI, the `managedResourceTags` field is inside the `properties` bag and the top level `tags`, `location`, etc fields are remaining. 
The motivation behind the field is:
1. the biggest reason is there are Azure policies that reject resource creation or do things about a resource based on the tags of the resource. Allowing customers to pass the tags for those resources created by RP is unblocking them from using our service or getting them off of a good start. 
2. the resources are visible to customers, but they are not created by customers but by RPs. customers have asked to be able to tag those multiple resources in a single place in API so that they don't need to tag various resources repeatedly every time.

Reply:
@feiyushi Thank you for the explanation. I have two more questions.
1. If there're Azure Policies to block certain operations, shouldn't them be enforced all the time? Is it justified to use this as a mechanism to bypass those policies?
2. Are `managedResourceTags` always in sync with the regular `tags` on resources? If so, how do you achieve that? If not, wouldn't it be confusing to the customers because they see two different sets of tags?

Reply:
> @feiyushi Thank you for the explanation. I have two more questions.
> 
> 1. If there're Azure Policies to block certain operations, shouldn't them be enforced all the time? Is it justified to use this as a mechanism to bypass those policies?
> 2. Are `managedResourceTags` always in sync with the regular `tags` on resources? If so, how do you achieve that? If not, wouldn't it be confusing to the customers because they see two different sets of tags?

1. how azure policy is enforced is dependent on how azure policy authored by the owner. sometimes they have "loopholes". it's the owners of those policies provide "opt-outs" via tags. this is a method chosen by them not by users. but it's a pattern.
2. `managedResourceTags` are totally independent with `tags` on the image template resource. `managedResourceTags` are tags on the AIB created vms, and other resources. so, there's no confusion here as long as customer understands what this `managedResourceTags` is targeted for. it's not two sets of tags on any single resource.